mvc/src/main/java/cz/muni/fi/mvc/controllers/ComponentController.java

```diff
@@ -0,0 +1,177 @@
+package cz.muni.fi.mvc.controllers;
+
+import cz.muni.fi.dto.ComponentDTO;
+import cz.muni.fi.filter.ComponentFilter;
+import cz.muni.fi.enums.ComponentType;
+import cz.muni.fi.facade.ComponentFacade;
+import cz.muni.fi.mvc.Tools;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.FieldError;
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.Valid;
+import java.util.Arrays;
+
+/**
+ * @author Robert Tamas
+ */
+@Controller
+@RequestMapping("/components")
+public class ComponentController {
+
+    private final static Logger log = LoggerFactory.getLogger(ComponentController.class);
+
+    @Autowired
+    ComponentFacade componentFacade;
+
+    @RequestMapping(value = "/create", method = RequestMethod.GET)
+    public String createComponent(Model model,  HttpServletRequest request, UriComponentsBuilder uriBuilder, RedirectAttributes redirectAttributes) {
+        //String res = Tools.redirectNonTestDriver(request, uriBuilder, redirectAttributes);
```

Comment:
I suppose we will use these commented parts in security part. Am I right?

Reply:
yep